src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModelUnitTest.java

```diff
@@ -106,38 +277,66 @@ public void setupModel() {
     }
 
     @Test(dataProvider = "CalcNIndelInformativeReadsData")
-    public void testCalcNIndelInformativeReads(final String readBases, final String cigar, final String ref, final int maxIndelSize, final List<Integer> expected ) {
+    public void testCalcNIndelInformativeReads(final String readBases, final String cigar, final byte[] readQuals, final String ref, final int maxIndelSize, final int readStartIntoRef, final List<Integer> expected ) {
         final byte qual = (byte)30;
-        final byte[] quals = Utils.dupBytes(qual, readBases.length());
+        final byte[] quals = readQuals != null ? readQuals : Utils.dupBytes(qual, readBases.length());
+        // on the same read after the first site the result will be cached in the transient attributes, assert the results are the same as those calculated non-transiently.
+        final GATKRead readCache = ArtificialReadUtils.createArtificialRead(readBases.getBytes(), quals, cigar);
 
         for ( int i = 0; i < readBases.getBytes().length; i++ ) {
-            final GATKRead read = ArtificialReadUtils.createArtificialRead(readBases.getBytes(), quals, cigar);
-            final SimpleInterval loc = new SimpleInterval("20", i + 1, i + 1);
-            final ReadPileup pileup = new ReadPileup(loc, Collections.singletonList(read), i);
-            final int actual = model.calcNIndelInformativeReads(pileup, i, ref.getBytes(), maxIndelSize);
-            Assert.assertEquals(actual, (int)expected.get(i), "failed at position " + i);
+            final Pair<Integer, Boolean> readCoordinateForReferenceCoordinate = ReadUtils.getReadCoordinateForReferenceCoordinate(readCache, readCache.getStart() + i, true);
+
+            if (!readCoordinateForReferenceCoordinate.getValue() && readCoordinateForReferenceCoordinate.getKey() != -1) {
+                final GATKRead readNoCache = ArtificialReadUtils.createArtificialRead(readBases.getBytes(), quals, cigar);
+                final SimpleInterval loc = new SimpleInterval("20", i + 1 + readStartIntoRef, i + 1 + readStartIntoRef);
+                final ReadPileup pileupCache = new ReadPileup(loc, Collections.singletonList(readCache), readCoordinateForReferenceCoordinate.getKey());
+                final ReadPileup pileupNoCache = new ReadPileup(loc, Collections.singletonList(readNoCache), ReadUtils.getReadCoordinateForReferenceCoordinate(readNoCache, readNoCache.getStart() + i).getKey());
+                final int actualCache = model.calcNIndelInformativeReads(pileupCache, i + readStartIntoRef, ref.getBytes(), maxIndelSize);
+                final int actualNoCache = model.calcNIndelInformativeReads(pileupNoCache, i + readStartIntoRef, ref.getBytes(), maxIndelSize);
+                Assert.assertEquals(actualCache, (int)expected.get(i), "cached result failed at position " + i);
+                Assert.assertEquals(actualNoCache, (int)expected.get(i), "non-cached result failed at position " + i);
+            }
         }
     }
```

Comment:
Can you confirm that you've tested this branch on full-scale data (whole genome or exome), and shown that the output at full scale hasn't changed?

Reply:
Ran this vs its root master branch. The data of the gvcf output files are md5 concordant on a battery of exomes.